bazel/win_flex_bison.bzl

```diff
@@ -0,0 +1,119 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Bazel rule to run flex/bison on Windows
+"""
+
+# buildifier: disable=provider-params
+FlexToolchainInfo = provider(
+    fields = [
+        "all_files",
+        "flex_tool",
+        "flex_lexer_h",
+    ],
+)
+
+def _win_flex_toolchain_impl(ctx):
+    flex_runfiles = ctx.attr.flex_tool[DefaultInfo].default_runfiles.files
+
+    return platform_common.ToolchainInfo(
+        flex_toolchain = FlexToolchainInfo(
+            all_files = depset(
+                direct = [ctx.executable.flex_tool],
+                transitive = [flex_runfiles],
+            ),
+            flex_tool = ctx.attr.flex_tool.files_to_run,
+            flex_lexer_h = ctx.file.flex_lexer_h,
+        ),
+    )
+
+win_flex_toolchain = rule(
+    _win_flex_toolchain_impl,
+    attrs = {
+        "flex_tool": attr.label(
+            mandatory = True,
+            executable = True,
+            cfg = "host",
+        ),
+        "flex_lexer_h": attr.label(
+            mandatory = True,
+            allow_single_file = [".h"],
+        ),
+    },
+)
+
+# buildifier: disable=provider-params
+BisonToolchainInfo = provider(
+    fields = [
+        "all_files",
+        "bison_tool",
+    ],
+)
+
+def _win_bison_toolchain_impl(ctx):
+    bison_runfiles = ctx.attr.bison_tool[DefaultInfo].default_runfiles.files
+
+    return platform_common.ToolchainInfo(
+        bison_toolchain = BisonToolchainInfo(
+            all_files = depset(
+                direct = [ctx.executable.bison_tool],
+                transitive = [bison_runfiles],
+            ),
+            bison_tool = ctx.attr.bison_tool.files_to_run,
+        ),
+    )
+
+win_bison_toolchain = rule(
+    _win_bison_toolchain_impl,
+    attrs = {
+        "bison_tool": attr.label(
+            mandatory = True,
+            executable = True,
+            cfg = "host",
+        ),
+    },
+)
+
+def _remote_win_flex_bison(repository_ctx):
+    repository_ctx.download_and_extract(
+        url = repository_ctx.attr.url,
+        sha256 = repository_ctx.attr.sha256,
+    )
+
+    repository_ctx.symlink(
+        Label("@com_google_verible//bazel:win_flex_bison.BUILD"),
+        "BUILD.bazel",
+    )
+
+remote_win_flex_bison = repository_rule(
+    implementation = _remote_win_flex_bison,
+    local = True,
+    attrs = {
+        "url": attr.string(mandatory = True),
+        "sha256": attr.string(default = ""),
+    },
+)
+
+def win_flex_configure(name, url, sha256 = ""):
+    remote_win_flex_bison(
```

Comment:
Could this `win_flex_configure()` be made to return early if we're not on Windows ?

There seems to be an issue with pre-fetching all the needed dependencies vs. what is then actually needed in download later in environments that separate out the fetching and building like on nixos; this is why it needs this patch:
https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/tools/language-servers/verible/remove-unused-deps.patch

I suspect what is happening: the `win_flex_configure()` called from the `WORKSPACE` unconditionally downloads the dependencies , but then on a non-Windows system, the `select()` in the `{bison,flex}.bzl` realizes that it doesn't actually needs it, so it then leaves it out of the prepeared download blob.
Later, after disconnect from the network, on `build`, the unconditional download attempt will fail. At least this is what I _think_ is happening.

Anyway, if here, we could just skip calling the `remote_win_flex_bison()` and register toolchain part if we're not on windows, that should probably help. Do you think you can add that @corco ?

Reply:
I'll will look at it at the end of the week

Reply:
I took a look. It seems that [this issue](https://github.com/bazelbuild/bazel/issues/13493) is part of the problem. There may be a way to fix it in either nix or bazel, but I am not comfortable with either...

I would suggest to either wait until bazel fix the issue, or simply remove downloading win_flex_bison and require the user to install it prior with chocolatey, like the use_local_flex_bison flag. I would move toward that second solution if @hzeller agrees.

Reply:
Oh, sorry, indeed I didn't see your previous comment.
I think removing the download and let the user `use_local_flex_bison` sounds good to me.